packages/devtools_app/lib/src/vm_developer/vm_service_private_extensions.dart

```diff
@@ -0,0 +1,93 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:meta/meta.dart';
+import 'package:vm_service/vm_service.dart';
+
+/// NOTE: this file contains extensions to classes provided by
+/// `package:vm_service` in order to expose VM internal fields in a controlled
+/// fashion. Objects and extensions in this class should not be used outside of
+/// the `vm_developer` directory.
+
+/// Convenience extension to create a [VMPrivateView] from a [VM].
+extension VMPrivateViewExtension on VM {
+  VMPrivateView toPrivateView() {
+    return this == null ? null : VMPrivateView._(this);
+  }
+}
+
+// TODO(bkonyi): figure out if we can properly extend VM.
+/// A wrapper around [VM] which allows for access to VM internal fields.
+class VMPrivateView {
+  VMPrivateView._(this.vm);
+
+  String get embedder => vm.json['_embedder'];
+  String get profilerMode => vm.json['_profilerMode'];
+  int get currentMemory => vm.json['_currentMemory'];
+  int get currentRSS => vm.json['_currentRSS'];
+  int get maxRSS => vm.json['_maxRSS'];
+  int get nativeZoneMemoryUsage => vm.json['_nativeZoneMemoryUsage'];
```

Comment:
is there a reason these can't be directly in VMPrivateViewExtension?

Reply:
I thought there was, but apparently that works just fine... Good suggestion; it really cleans things up!